Subject: [PATCH] [ActivityManager] fix service could not be started correctly when app.thread == null

there is not thread object attached to application when starting service,
will not start process, so make start service failed.

startActivity, broadcast or  provider all have condition:
 if app == null || app.thread == null,  that will start process.
but startService only if app == null, lack of  condition app.thread == null,  that will start process.

diff --git a/services/core/java/com/android/server/am/ActiveServices.java b/services/core/java/com/android/server/am/ActiveServices.java
index d914881..af26b0d 100644
--- a/services/core/java/com/android/server/am/ActiveServices.java
+++ b/services/core/java/com/android/server/am/ActiveServices.java
@@ -1800,7 +1800,7 @@ private String bringUpServiceLocked(ServiceRecord r, int intentFlags, boolean ex
 
         // Not running -- get it started, and enqueue this service record
         // to be executed when the app comes up.
-        if (app == null && !permissionsReviewRequired) {
+        if ((app == null || app.thread == null) && !permissionsReviewRequired) {
             if ((app=mAm.startProcessLocked(procName, r.appInfo, true, intentFlags,
                     "service", r.name, false, isolated, false)) == null) {
                 String msg = "Unable to launch app "
